SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+
+The LSST data processing pipeline software has been in development as an open source project since 2004\cite{2004AAS...20510811A,2010SPIE.7740E..15A,2016_adassxxv_P056} and during these 16 years of development much has changed in the Python, C++ and astronomy software world.
+
+
+\section{The LSST Science Pipeline Software}
+
+As with other software developed in the early 2000s, such as CASA\cite{2012ASPC..461..849P}, the LSST science pipeline software is based on a C++ core with a Python control layer, where the Python/C++ interface layer is handled via SWIG\cite{beazley2003automated} bindings.
+The general layout of the code is shown in Fig.~\ref{fig:layers}.
+At the time of writing the LSST science pipelines codebase is split between C++ and Python with of order 105,000 lines of Python and 125,000 lines of C++\footnote{This total does not include comment or blank lines, nor does it include the 80,000 lines of Python and 1.6\,million lines of C++ generated for the SWIG wrappers.}.
+
+\begin{figure} [t]
+\begin{center}
+\includegraphics[height=5cm]{Software-Layers}
+\end{center}
+\caption[layers]
+%>>>> use \label inside caption to get Fig. number with \ref{}
+{\label{fig:layers}
+Structure of the LSST data management software.
+At the bottom there are external packages (black), on the lower left (red) are the computationally intensive packages that are mainly written in C++ with SWIG wrappers making the code available in Python, and the remainder (blue) are mostly Python.}
+\end{figure}
+
+The software is spread among many different root packages within the \texttt{lsst} namespace:
+
+\begin{description}
+\item[\textbf{afw}] These provide core LSST functionality including geometric shapes, images, source detection, convolution and high-level exposure objects.
+
+\item[\textbf{daf}] The data access framework, also known as \emph{the Data Butler}, provides an abstraction layer hiding data formats and locations from the user and providing them with standardized Python objects representing the data.
+
+\item[\textbf{ip}] Image processing algorithms such as instrument signature removal and image differencing.
+
+\item[\textbf{meas}] Measurement algorithms.
+
+\item[\textbf{obs}] Observatory plugins to teach the system how to handle files from different instruments.
+
+\item[\textbf{pipe}] Pipeline processing infrastructure. This allows discrete units of computation to be linked together to form pipelines.
+
+\end{description}
+
+In terms of overlap with Astropy and affiliate packages the AFW and \texttt{meas} packages are most relevant for immediate investigation.
+
+
+\section{Astropy}
+
+Python has become the de facto standard programming language for astronomy\cite{2000ASPC..216...59G,2006ASPC..351..343H,2006ASPC..351..497K,2011ASPC..442..425G,2012SPIE.8451E..02J} and is now the default choice for new projects and for teaching.
```

Comment:
Maybe just "Python has become the dominant programming language for astronomy", then?  "de facto standard" also seems a little too strong to me, as it seems to imply that other choices are now incorrect, rather than just less popular.